node.js

```diff
@@ -266,122 +331,150 @@ async function getSuccessor(id, node_querying, node_queried) {
         const node_queried_client = caller(`localhost:${node_queried.port}`, PROTO_PATH, "Node");
         // now grab the remote value
         try {
-            n_successor = await node_queried_client.getSuccessor_remote(node_queried);
+            n_successor = await node_queried_client.getSuccessor_remotehelper(node_queried);
         } catch (err) {
-            console.error("getSuccessor_remote error in getSuccessor() ", err);
+            // TBD 20191103.hk: why does "n_successor = NULL_NODE;" not do the same as explicit?!?!
+            n_successor = {id: null, ip: null, port: null};
+            if (DEBUGGING_LOCAL) {
+                console.trace("Remote error in getSuccessor() ", err);
+            }
         }
     }
 
-    /* vvvvv     vvvvv     TBD debugging code     vvvvv     vvvvv */
     if (DEBUGGING_LOCAL) {
-        console.log("returning n_successor = ", n_successor);
+        console.log("returning {", node_queried.id, "}.successor = ", n_successor.id);
         console.log("^^^^^     ^^^^^     getSuccessor     ^^^^^     ^^^^^");
     }
-    /* ^^^^^     ^^^^^     TBD debugging code     ^^^^^     ^^^^^ */
 
     return n_successor;
 }
 
-/* modified 20191102 */
+/** 
+ * RPC equivalent of the getSuccessor() method.
+ * It is implemented as simply a wrapper for the getSuccessor() function.
+ * 
+ * @version 20191103
+ */
+async function getSuccessor_remotehelper(thing, callback) {
+    callback(null, FingerTable[0].successor);
+}
+
+/**
+ * Directly implement the pseudocode's closest_preceding_finger() method.
+ * 
+ * However, it is able to discern whether to do a local lookup or an RPC.
+ * If the querying node is the same as the queried node, it will stay local.
+ * 
+ * @returns the closest preceding node to ID
+ * 
+ * @version 20191103
+ */
 async function closest_preceding_finger(id, node_querying, node_queried) {
-    /**
-     * Directly implement the pseudocode's closest_preceding_finger() method.
-     * However, it is able to discern whether to do a local lookup or an RPC.
-     * If the querying node is the same as the queried node, it will stay local.
-     */
-  let n_preceding;
-  if (node_querying.id == node_queried.id) {
-    // use local value
-    // for i = m downto 1
-    for (let i = HASH_BIT_LENGTH - 1; i >= 0; i--) {
-      // if ( finger[i].node 'is-in' (n, id) )
-      if ( isInModuloRange(FingerTable[i].successor.id, node_queried.id, false, id, false) ) {
-        // return finger[i].node;
-        n_preceding = FingerTable[i].successor;
+    let n_preceding = NULL_NODE;
+    if (node_querying.id == node_queried.id) {
+        // use local value
+        // for i = m downto 1
```

Comment:
The comments in this function seem a bit excessive to me. What is your intention here? Do they tied to the pseudocode or something?